### PR TITLE
test: migrate rename_deps, replace, required_features and run to snapbox

### PR DESCRIPTION
**Pull request by eth3lbert**
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/rename_deps.rs`
- `tests/testsuite/replace.rs`
- `tests/testsuite/required_features.rs`
- `tests/testsuite/run.rs`

### CODE REVIEW

File: tests/testsuite/required_features.rs [hunk 1]
```diff
@@ -399,29 +419,54 @@ fn test_multiple_required_features() {
...
 
     p.cargo("test --features c")
-        .with_stderr(
-            "\
-[COMPILING] foo v0.0.1 ([CWD])
-[FINISHED] `test` profile [unoptimized + debuginfo] target(s) in [..]
-[RUNNING] [..] (target/debug/deps/foo_1-[..][EXE])
-[RUNNING] [..] (target/debug/deps/foo_2-[..][EXE])",
-        )
-        .with_stdout_contains_n("test test ... ok", 2)
```
Comment (eth3lbert) [inline tests/testsuite/required_features.rs:419]: The `with_stdout_contains_n` function should be analogous to `with_stdout_contains` where the expected output appears n times, and the tracking issue doesn't mention this function.

File: tests/testsuite/required_features.rs [hunk 2]
```diff
@@ -590,29 +661,54 @@ fn bench_multiple_required_features() {
...
 
     p.cargo("bench --features c")
-        .with_stderr(
-            "\
-[COMPILING] foo v0.0.1 ([CWD])
-[FINISHED] `bench` profile [optimized] target(s) in [..]
-[RUNNING] [..] (target/release/deps/foo_1-[..][EXE])
-[RUNNING] [..] (target/release/deps/foo_2-[..][EXE])",
-        )
-        .with_stdout_contains_n("test bench ... bench: [..]", 2)
```
Comment (eth3lbert) [inline tests/testsuite/required_features.rs:610]: ditto.

File: tests/testsuite/replace.rs [hunk 3]
```diff
@@ -473,22 +460,21 @@ fn use_a_spec_to_select() {
...
-[CHECKING] [..]
-[CHECKING] [..]
-[CHECKING] [..]
-[CHECKING] foo v0.0.1 ([CWD])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
-",
-        )
+[DOWNLOADED] baz v0.1.1 (registry `dummy-registry`)
+[DOWNLOADED] bar v0.1.1 (registry `dummy-registry`)
+[CHECKING] baz v0.2.0 ([ROOTURL]/override#[..])
```
[the remark this answers is not in the document]
Comment (eth3lbert) [reply tests/testsuite/replace.rs:471]: Yes, it is.
FYI:
```
...
[CHECKING] baz v0.2.0 ([ROOTURL]/override#15da3cb8)
...
```

[the remark this answers is not in the document]
Comment (epage) [reply tests/testsuite/replace.rs:471]: Hmm, not seeing a good way to auto-redact that.  Oh well.

**Pull request merged**